bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java

```diff
@@ -144,6 +145,26 @@ private boolean isAppInForeground() {
         return mActivityCount > 0 || mIsConfigChange;
     }
 
+    /**
+     * If it's a fresh start or back to main activity, we can safely clear cache
+     */
+    private boolean isFreshStartOrBackToMain(@NonNull Activity activity, Bundle savedInstanceState) {
+        if (savedInstanceState != null) {
+            return false;
+        }
+        ActivityManager systemService = (ActivityManager) activity.getSystemService(Context.ACTIVITY_SERVICE);
+        if (systemService == null) {
+            return false;
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            List<ActivityManager.AppTask> appTasks = systemService.getAppTasks();
+            return appTasks != null && appTasks.size() == 1 && appTasks.get(0).getTaskInfo().numActivities == 1;
```

Comment:
I'm pretty sure its not possible for `appTasks` to be `null` here. If you can provide documentation or code that shows otherwise we can keep the null check, otherwise let's remove it. Same thing with the check on `runningTasks` below.